### ISSUE TITLE
AWS::StepFunctions::StateMachine - `E3601` is unable to specify the branch for `Parallel` states and correctly display the error location

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4311]
### CloudFormation Lint Version

cfn-lint 1.42.0

### What operating system are you using?

Windows 11

### Describe the bug

While checking https://github.com/aws-cloudformation/cfn-lint/pull/4308 i noticed that cfn-lint currently cannot specify the branch in a `Parallel` state for `StartAt` issues and display the right location in the VS Code extension (although it is able to find the issue):

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  Role:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt Role.Arn
      Definition:
        QueryLanguage: JSONPath
        StartAt: ParallelExecution
        States: # [cfn-lint] E3601: Missing 'Next' target 'FAIL' at /States/TestMap/ItemProcessor/StartAt
          ParallelExecution:
            Type: Parallel
            Branches:
              - StartAt: TestPass
                States:
                  TestPass:
                    Type: Pass
                    End: true
              - StartAt: TestMap
                States:
                  TestMap:
                    Type: Map
                    ItemsPath: $
                    ItemProcessor:
                      ProcessorConfig:
                        Mode: INLINE
                      StartAt: FAIL # AWS: MISSING_TRANSITION_TARGET: Missing 'Next' target: FAIL at /States/ParallelExecution/Branches[1]/States/TestMap/ItemProcessor/StartAt
                      States:
                        TestPassTwo:
                          Type: Pass
                          End: true
                    End: true
            End: true
```

### Expected behavior

cfn-lint's error and VS Code error location should match the Step Function API.

### Reproduction template

See above
